### PR TITLE
Use mypy in our CI script

### PR DESCRIPTION
**Pull request by selitvin**
Using mypy to run static type checking as part of travis build

Fixed some issues causing mypy to fail:
- Upgraded minimal torch and Pillow versions in `setup.py`
- Upgraded torch and Pillow in `.travis.yml`. This is a quick fix. A proper better way is to upgrade docker image used for the build.
- Rely on `unittest`'s `mock`
- Adding some type hints required by mypy

### CODE REVIEW

File: mypy.ini [hunk 1]
```diff
@@ -0,0 +1,5 @@
+[mypy]
+ignore_missing_imports = True
```
Comment (jsgoller1) [inline mypy.ini:2]: Do we want this to be true? Is there a particular case where this causes some kind of false positive?

File: .travis.yml [hunk 2]
```diff
@@ -57,10 +57,15 @@ before_script:
   - $RUN pip install --upgrade numpy
 
   - $RUN pip install -U pyarrow==${PYARROW_VERSION} tensorflow==${TF_VERSION}
+
+  # TODO(yevgeni): need to update docker build image at some stage
+  - $RUN pip install -U torchvision==0.2.1 torch==1.5.1 Pillow==7.1.2
```
Comment (jsgoller1) [inline .travis.yml:62]: Can we do this [in the Dockerfile](https://github.com/uber/petastorm/blob/master/docker/Dockerfile#L34)? Should be a really quick one-line change, I think. Possibly saves us a nasty headache later on if trying to debug package issues.

File: mypy.ini (hunk 1 above)
Comment (selitvin) [reply mypy.ini:2]: I tried listing external libraries one-by-one when they failed mypy for not good reasons. Gave up eventually.

File: .travis.yml (hunk 2 above)
Comment (selitvin) [reply .travis.yml]: Done. Going to land the PR.

File: setup.py [hunk 3]
```diff
@@ -33,6 +33,7 @@
     'dill>=0.2.1',
     'diskcache>=3.0.0',
     'future>=0.10.2',
+    'gcsfs>=0.2.0',
```
Comment (mengxr) [inline setup.py:36]: @selitvin Is `gcsfs` required?

**Pull request merged**